app/src/main/java/com/cornellappdev/scoop/ui/login/AuthScreen.kt

```diff
@@ -0,0 +1,172 @@
+package com.cornellappdev.scoop.ui.login
+
+import android.app.Activity
+import android.util.Log
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.cornellappdev.scoop.R
+import com.cornellappdev.scoop.data.repositories.LoginRepository
+import com.cornellappdev.scoop.data.repositories.SignedInUser
+import com.cornellappdev.scoop.ui.navigation.MainScreen
+import com.cornellappdev.scoop.ui.theme.Green
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.tasks.Task
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Response
+import org.json.JSONObject
+import java.io.IOException
+
+
+@Composable
+fun AuthFlow(signInClient: GoogleSignInClient) {
+
+    val navController = rememberNavController()
+
+    // Set up the NavHost with the navigation graph
+    NavHost(navController = navController, startDestination = "AUTH") {
+        composable("AUTH") {
+            AuthScreen(signInClient, navController)
+        }
+        composable("HOME") {
+            MainScreen()
+
+        }
+    }
+
+}
```

Comment:
Perhaps rename to something like `AuthNav` or something to be a bit more clear... and add a spec 😉